torchtune/training/checkpointing/_utils.py

```diff
@@ -277,3 +299,161 @@ def update_state_dict_for_classifier(
         state_dict.pop("output.bias")
     if state_dict["output.weight"].shape[0] != output_weight.shape[0] or force_override:
         state_dict["output.weight"] = output_weight
+
+
+# TODO: add test
+def get_largest_iter_folder(dir: Union[str, Path], pattern: str = r"^epoch_(\d+)"):
+    latest_epoch_folder = None
+    iter_folders = []
+    regex = re.compile(pattern)
+
+    # Iterate over the directory contents
+    for fname in os.listdir(dir):
+        match = regex.match(fname)
+        if match:
+            # Extract the number from the match
+            iter_number = int(match.group(1))
+            iter_folders.append((fname, iter_number))
+
+    # Find the folder with the largest iter number
+    if iter_folders:
+        latest_epoch_folder = max(iter_folders, key=lambda x: x[1])[0]
+
+    return latest_epoch_folder
+
+
+# TODO: instead of copying, make it a symlink when we start using HF cache
+def copy_files(
```

Comment:
Can you make ignore suffixes kwarg only?

Reply:
I don't follow. Can you give an example?

Reply:
```python
def copy_files(
	input,
	output,
	*,
	ignore_suffixes
)
```

Reply:
Call me crazy but do we need a full Python function reinventing recursive copy here? E.g. 

```
os.system(
f"rsync -av --ignore-existing {" ".join([f"--exclude *{}" for ignore_suffix in ignore_suffixes])} {input_dir} {output_dir}"
)
```

Reply:
I just use [shutil.copy_tree](https://docs.python.org/3/library/shutil.html#shutil.copytree)

Reply:
Yeah wait actually is there a reason why can't use copytree? I'd rather that be maintained by core Python.

Reply:
beyond the suffixes, ii will also have to ignore .cache and .git__

idk, the function gives us some flexibility and its readable. But it seems to be 3x1. 

Can we leave it for the ckpt refactoring?

Reply:
Hmm I do like copy_tree and feel like it should be workable. But won't block on it, just include it in the follow-up task (as I mentioned above)